frames/src/main/java/com/checkout/frames/screen/billingaddress/billingaddressdetails/BillingAddressDetailsViewModel.kt

```diff
@@ -0,0 +1,133 @@
+
+@Suppress("LongParameterList", "UnusedPrivateMember")
+internal class BillingAddressDetailsViewModel @Inject constructor(
+    private val paymentStateManager: PaymentStateManager,
+    private val textLabelStyleMapper: Mapper<TextLabelStyle, TextLabelViewStyle>,
+    private val textLabelStateMapper: Mapper<TextLabelStyle?, TextLabelState>,
+    private val containerMapper: Mapper<ContainerStyle, Modifier>,
+    private val imageMapper: ImageStyleToDynamicComposableImageMapper,
+    private val billingAddressDetailsComponentStateUseCase:
+    UseCase<List<BillingAddressInputComponentStyle>, BillingAddressInputComponentsContainerState>,
+    private val billingAddressDetailsComponentStyleUseCase:
+    UseCase<List<BillingAddressInputComponentStyle>, List<BillingAddressInputComponentViewStyle>>,
+    private val buttonStyleMapper: Mapper<ButtonStyle, InternalButtonViewStyle>,
+    private val buttonStateMapper: Mapper<ButtonStyle, InternalButtonState>,
+    private val style: BillingAddressDetailsStyle
+) : ViewModel() {
+    // Header styling
+    val screenTitleStyle = textLabelStyleMapper.map(style.billingAddressHeaderComponentStyle.headerTitleStyle)
+    val screenTitleState = provideScreenTitleState(style.billingAddressHeaderComponentStyle.headerTitleStyle)
+    val screenModifier = containerMapper.map(style.containerStyle)
+
+    val screenButtonState = provideButtonState()
+    val screenButtonStyle = buttonStyleMapper.map(style.billingAddressHeaderComponentStyle.headerButtonStyle)
+
+    val goBack = mutableStateOf(false)
+
+    val billingAddressInputComponentsContainerModifier =
+        containerMapper.map(style.billingAddressInputComponentsContainerStyle.containerStyle)
+
+    val billingAddressInputComponentState = provideBillingAddressInputComponentsState(
+        style.billingAddressInputComponentsContainerStyle
+    )
+    val billingAddressInputComponentsStyle = provideBillingAddressInputComponentsStyle(
+        style.billingAddressInputComponentsContainerStyle
+    )
+
+    private fun provideBillingAddressInputComponentsStyle(
+        billingAddressInputComponentsContainerStyle: BillingAddressInputComponentsContainerStyle
+    ): List<BillingAddressInputComponentViewStyle> {
+        return billingAddressDetailsComponentStyleUseCase.execute(
+            billingAddressInputComponentsContainerStyle.billingAddressInputComponentStyleList
+        )
+    }
+
+    private fun provideBillingAddressInputComponentsState(
+        billingAddressInputComponentsContainerStyle:
+        BillingAddressInputComponentsContainerStyle
+    ): BillingAddressInputComponentsContainerState {
+        return billingAddressDetailsComponentStateUseCase.execute(
+            billingAddressInputComponentsContainerStyle.billingAddressInputComponentStyleList
+        )
+    }
+
+    fun onFocusChanged(isFocused: Boolean) {
+        Log.d("", "" + isFocused)
+    }
+
+    fun onSearchChange(text: String) {
+        Log.d("", "" + text)
+    }
```

Comment:
Is it still needed?

Reply:
Yes, it is part of the next PR. A temporary log is provided to avoid Lint Gradle task failure.